### PR TITLE
Fix #2413: Add note to docs for creating sn project in mounted directories

### LINKED ISSUE #2413: ScalaNative Linking issue when running with wsl (windows subsystem linux)
I am facing error when I run `sbt run` on gitter8 template provided by scala-native.

**Error**
```
sbt:javatimeexp> run
[info] Linking (797 ms)
[info] Discovered 971 classes and 7868 methods
[info] Optimizing (release-fast mode) (3201 ms)
[error] stack trace is suppressed; run last Compile / nativeLink for the full output
[error] (Compile / nativeLink) java.io.IOException: No such file or directory
[error] Total time: 6 s, completed Oct 20, 2021 10:01:58 AM
```

```
sbt:javatimeexp> last Compile / nativeLink
[info] Linking (907 ms)
[info] Discovered 971 classes and 7868 methods
[info] Optimizing (release-fast mode) (3338 ms)
[error] java.io.IOException: No such file or directory
[error]         at sun.nio.ch.FileDispatcherImpl.size0(Native Method)
[error]         at sun.nio.ch.FileDispatcherImpl.size(FileDispatcherImpl.java:88)
[error]         at sun.nio.ch.FileChannelImpl.size(FileChannelImpl.java:310)
[error]         at scala.scalanative.io.VirtualDirectory$NioDirectory.$anonfun$merge$1(VirtualDirectory.scala:123)
[error]         at scala.scalanative.io.VirtualDirectory$NioDirectory.$anonfun$merge$1$adapted(VirtualDirectory.scala:118)
[error]         at scala.collection.immutable.List.foreach(List.scala:431)
[error]         at scala.scalanative.io.VirtualDirectory$NioDirectory.merge(VirtualDirectory.scala:118)
[error]         at scala.scalanative.io.VirtualDirectory$NioDirectory.merge$(VirtualDirectory.scala:112)
[error]         at scala.scalanative.io.VirtualDirectory$LocalDirectory.merge(VirtualDirectory.scala:130)
[error]         at scala.scalanative.codegen.CodeGen$Impl.gen(CodeGen.scala:108)
[error]         at scala.scalanative.codegen.CodeGen$.$anonfun$emit$5(CodeGen.scala:61)
[error]         at scala.collection.parallel.AugmentedIterableIterator.map2combiner(RemainsIterator.scala:116)
[error]         at scala.collection.parallel.AugmentedIterableIterator.map2combiner$(RemainsIterator.scala:113)
[error]         at scala.collection.parallel.immutable.ParHashMap$ParHashMapIterator.map2combiner(ParHashMap.scala:80)
[error]         at scala.collection.parallel.ParIterableLike$Map.leaf(ParIterableLike.scala:1064)
[error]         at scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:53)
[error]         at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error]         at scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:67)
[error]         at scala.collection.parallel.Task.tryLeaf(Tasks.scala:56)
[error]         at scala.collection.parallel.Task.tryLeaf$(Tasks.scala:50)
[error]         at scala.collection.parallel.ParIterableLike$Map.tryLeaf(ParIterableLike.scala:1061)
[error]         at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.internal(Tasks.scala:160)
[error]         at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.internal$(Tasks.scala:157)
[error]         at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.internal(Tasks.scala:440)
[error]         at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute(Tasks.scala:150)
[error]         at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute$(Tasks.scala:149)
[error]         at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:440)
[error]         at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
[error]         at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
[error]         at java.util.concurrent.ForkJoinTask.doJoin(ForkJoinTask.java:389)
[error]         at java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:719)
[error]         at scala.collection.parallel.ForkJoinTasks$WrappedTask.sync(Tasks.scala:379)
[error]         at scala.collection.parallel.ForkJoinTasks$WrappedTask.sync$(Tasks.scala:379)
[error]         at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.sync(Tasks.scala:440)
[error]         at scala.collection.parallel.ForkJoinTasks.executeAndWaitResult(Tasks.scala:423)
[error]         at scala.collection.parallel.ForkJoinTasks.executeAndWaitResult$(Tasks.scala:416)
[error]         at scala.collection.parallel.ForkJoinTaskSupport.executeAndWaitResult(TaskSupport.scala:60)
[error]         at scala.collection.parallel.ExecutionContextTasks.executeAndWaitResult(Tasks.scala:555)
[error]         at scala.collection.parallel.ExecutionContextTasks.executeAndWaitResult$(Tasks.scala:555)
[error]         at scala.collection.parallel.ExecutionContextTaskSupport.executeAndWaitResult(TaskSupport.scala:84)
[error]         at scala.collection.parallel.ParIterableLike$ResultMapping.leaf(ParIterableLike.scala:968)
[error]         at scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:53)
[error]         at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error]         at scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:67)
[error]         at scala.collection.parallel.Task.tryLeaf(Tasks.scala:56)
[error]         at scala.collection.parallel.Task.tryLeaf$(Tasks.scala:50)
[error]         at scala.collection.parallel.ParIterableLike$ResultMapping.tryLeaf(ParIterableLike.scala:963)
[error]         at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute(Tasks.scala:153)
[error]         at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute$(Tasks.scala:149)
[error]         at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:440)
[error]         at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
[error]         at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
[error]         at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
[error]         at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
[error]         at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175)
[error] (Compile / nativeLink) java.io.IOException: No such file or directory
```
Setup
- Windows 10
- wsl env (Ubuntu 20.04 LTS)
- Clang, llvm & libgc-dev  installed as per documentation
- sbt new scala-native/scala-native.g8

Steps to Reproduce
- sbt new scala-native/scala-native.g8
- sbt run

### PR DESCRIPTION
**Pull request by ankusharya**
This commit includes
1. Note, when creating sn project in mounted directory
2. Add bash scripts to generate docs and add documention on running the bash script in `contrib/build.rst`

### CODE REVIEW

File: docs/user/sbt.rst [hunk 1]
```diff
@@ -13,6 +13,11 @@ template.  In an empty working directory, execute::
 
     sbt new scala-native/scala-native.g8
 
+.. note:: New project should not be created in `mounted` directories, like external storage devices.
+
+  Incase of WSL2(Windows Subsystem Linux), windows file system like `C` or `D` drive are percieved as `mounted`. So creating new projects in these location will not work.
```
Comment (WojciechMazur) [inline docs/user/sbt.rst:18]:
```suggestion
  In the case of WSL2 (Windows Subsystem Linux), Windows file system drives like `C` or `D` are perceived as `mounted`. So creating new projects in these locations will not work.
```

File: docs/user/sbt.rst [hunk 2]
```diff
@@ -13,6 +13,11 @@ template.  In an empty working directory, execute::
...
+
+  In wsl2 environment, creating new project wsl2 file path work. e.g /home/<USER>/sn-projects.
```
Comment (WojciechMazur) [inline docs/user/sbt.rst:20]:
```suggestion
  In the WSL2 environment, it is recommended to create projects in the user files path, e.g /home/<USER>/sn-projects.
```

File: scripts/makedocs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+function generate_docs {
+  make html
+}
+
+function usage {
+  echo "  Usage:"
+  echo "    1. bash makedocs setup # if setup is not done."
+  echo "    2. bash makedocs       # if setup is done and building docs is needed"
+
```
Comment (WojciechMazur) [inline scripts/makedocs:24]: Redundant blank line

**Review by WojciechMazur — COMMENTED** [review]
Thank you for your contribution, I've left some suggestions on documentation text which in same seemed a bit confusing to me or contained typos

**Comment by ankusharya** [comment]
@WojciechMazur Hi, I have made corrections as per your suggestion. Please review once.

**Review by WojciechMazur — APPROVED** [review]

**Comment by ekrich** [comment]
@ankusharya Congrats on your PR!

**Pull request merged**